Unable to `pip install multidict` with `gcc-14`

## Long story short

With the latest Fedora Rawhide which freshly contains `gcc-14` I'm not able to install `multidict`.

## Expected behaviour

Package successfully installed.

## Actual behaviour

I see the following error:

```
gcc -fno-strict-overflow -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fPIC -I/tmp/venv/include -I/usr/include/python3.12 -c multidict/_multidict.c -o build/temp.linux-x86_64-cpython-312/multidict/_multidict.o -O2 -std=c99 -Wall -Wsign-compare -Wconversion -fno-strict-aliasing -pedantic
multidict/_multidict.c: In function ‘multidict_getall’:
multidict/_multidict.c:458:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  458 |     static _PyArg_Parser _parser = {NULL, _keywords, "getall", 0};
      |                                     ^~~~
multidict/_multidict.c:458:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:458:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  458 |     static _PyArg_Parser _parser = {NULL, _keywords, "getall", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:458:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:458:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  458 |     static _PyArg_Parser _parser = {NULL, _keywords, "getall", 0};
      |                                                      ^~~~~~~~
multidict/_multidict.c:458:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_getone’:
multidict/_multidict.c:503:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  503 |     static _PyArg_Parser _parser = {NULL, _keywords, "getone", 0};
      |                                     ^~~~
multidict/_multidict.c:503:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:503:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  503 |     static _PyArg_Parser _parser = {NULL, _keywords, "getone", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:503:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:503:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  503 |     static _PyArg_Parser _parser = {NULL, _keywords, "getone", 0};
      |                                                      ^~~~~~~~
multidict/_multidict.c:503:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_get’:
multidict/_multidict.c:538:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  538 |     static _PyArg_Parser _parser = {NULL, _keywords, "get", 0};
      |                                     ^~~~
multidict/_multidict.c:538:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:538:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  538 |     static _PyArg_Parser _parser = {NULL, _keywords, "get", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:538:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:538:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  538 |     static _PyArg_Parser _parser = {NULL, _keywords, "get", 0};
      |                                                      ^~~~~
multidict/_multidict.c:538:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_tp_dealloc’:
multidict/_multidict.c:712:5: warning: ‘UsingDeprecatedTrashcanMacro’ is deprecated [-Wdeprecated-declarations]
  712 |     Py_TRASHCAN_SAFE_BEGIN(self);
      |     ^~~~~~~~~~~~~~~~~~~~~~
multidict/_multidict.c: In function ‘multidict_add’:
multidict/_multidict.c:780:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  780 |     static _PyArg_Parser _parser = {NULL, _keywords, "add", 0};
      |                                     ^~~~
multidict/_multidict.c:780:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:780:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  780 |     static _PyArg_Parser _parser = {NULL, _keywords, "add", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:780:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:780:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  780 |     static _PyArg_Parser _parser = {NULL, _keywords, "add", 0};
      |                                                      ^~~~~
multidict/_multidict.c:780:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_setdefault’:
multidict/_multidict.c:839:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  839 |     static _PyArg_Parser _parser = {NULL, _keywords, "setdefault", 0};
      |                                     ^~~~
multidict/_multidict.c:839:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:839:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  839 |     static _PyArg_Parser _parser = {NULL, _keywords, "setdefault", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:839:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:839:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  839 |     static _PyArg_Parser _parser = {NULL, _keywords, "setdefault", 0};
      |                                                      ^~~~~~~~~~~~
multidict/_multidict.c:839:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_popone’:
multidict/_multidict.c:875:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  875 |     static _PyArg_Parser _parser = {NULL, _keywords, "popone", 0};
      |                                     ^~~~
multidict/_multidict.c:875:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:875:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  875 |     static _PyArg_Parser _parser = {NULL, _keywords, "popone", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:875:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:875:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  875 |     static _PyArg_Parser _parser = {NULL, _keywords, "popone", 0};
      |                                                      ^~~~~~~~
multidict/_multidict.c:875:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_pop’:
multidict/_multidict.c:922:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  922 |     static _PyArg_Parser _parser = {NULL, _keywords, "pop", 0};
      |                                     ^~~~
multidict/_multidict.c:922:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:922:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  922 |     static _PyArg_Parser _parser = {NULL, _keywords, "pop", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:922:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:922:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  922 |     static _PyArg_Parser _parser = {NULL, _keywords, "pop", 0};
      |                                                      ^~~~~
multidict/_multidict.c:922:54: note: (near initialization for ‘_parser.keywords’)
multidict/_multidict.c: In function ‘multidict_popall’:
multidict/_multidict.c:970:37: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
  970 |     static _PyArg_Parser _parser = {NULL, _keywords, "popall", 0};
      |                                     ^~~~
multidict/_multidict.c:970:37: note: (near initialization for ‘_parser.initialized’)
multidict/_multidict.c:970:43: error: initialization of ‘const char *’ from incompatible pointer type ‘const char * const*’ [-Wincompatible-pointer-types]
  970 |     static _PyArg_Parser _parser = {NULL, _keywords, "popall", 0};
      |                                           ^~~~~~~~~
multidict/_multidict.c:970:43: note: (near initialization for ‘_parser.format’)
multidict/_multidict.c:970:54: error: initialization of ‘const char * const*’ from incompatible pointer type ‘char *’ [-Wincompatible-pointer-types]
  970 |     static _PyArg_Parser _parser = {NULL, _keywords, "popall", 0};
      |                                                      ^~~~~~~~
multidict/_multidict.c:970:54: note: (near initialization for ‘_parser.keywords’)
error: command '/usr/bin/gcc' failed with exit code 1
```

## Steps to reproduce

Install the [latest rawhide](https://kojipkgs.fedoraproject.org/compose/rawhide/Fedora-Rawhide-20240117.n.0/compose/Cloud/x86_64/images/Fedora-Cloud-Base-Rawhide-20240117.n.0.x86_64.qcow2).

```shell
python3 -m venv /tmp/venv
/tmp/venv/bin/pip install multidict
```

## Your environment

```
Fedora-Cloud-Base-Rawhide-20240117.n.0.x86_64.qcow2
gcc-14.0.1-0.1.fc40.x86_64
```

--- a/CHANGES/909.bugfix.rst
+++ b/CHANGES/909.bugfix.rst
@@ -1,3 +1,10 @@
-Revert to using the public argument parsing API
-:c:func:`PyArg_ParseTupleAndKeywords` under Python 3.13
--- :user:`webknjaz`.
+Reverted to using the public argument parsing API
+:c:func:`PyArg_ParseTupleAndKeywords` under Python 3.12
+-- by :user:`charles-dyfis-net` and :user:`webknjaz`.
+
+The effect is that this change prevents build failures with
+clang 16.9.6 and gcc-14 reported in :issue:`926`. It also
+fixes a segmentation fault crash caused by passing keyword
+arguments to :py:meth:`MultiDict.getall()
+<multidict.MultiDict.getall>` discovered by :user:`jonaslb`
+and :user:`hroncok` while examining the problem.

--- a/CHANGES/926.bugfix.rst
+++ b/CHANGES/926.bugfix.rst
@@ -1,0 +1,1 @@
+909.bugfix.rst

--- a/CHANGES/929.bugfix.rst
+++ b/CHANGES/929.bugfix.rst
@@ -1,0 +1,1 @@
+909.bugfix.rst

--- a/docs/spelling_wordlist.txt
+++ b/docs/spelling_wordlist.txt
@@ -33,6 +33,7 @@ eof
 fallback
 fastpath
 filename
+gcc
 getitem
 github
 google

--- a/multidict/_multidict.c
+++ b/multidict/_multidict.c
@@ -9,7 +9,7 @@
 #include "_multilib/iter.h"
 #include "_multilib/views.h"
 
-#if PY_MAJOR_VERSION < 3 || PY_MINOR_VERSION < 13
+#if PY_MAJOR_VERSION < 3 || PY_MINOR_VERSION < 12
 #ifndef _PyArg_UnpackKeywords
 #define FASTCALL_OLD
 #endif
@@ -444,7 +444,7 @@ fail:
 static inline PyObject *
 multidict_getall(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -458,7 +458,7 @@ multidict_getall(
              *key      = NULL,
              *_default = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *getall_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:getall",
@@ -509,7 +509,7 @@ skip_optional_pos:
 static inline PyObject *
 multidict_getone(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -522,7 +522,7 @@ multidict_getone(
     PyObject *key      = NULL,
              *_default = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *getone_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:getone",
@@ -563,7 +563,7 @@ skip_optional_pos:
 static inline PyObject *
 multidict_get(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -577,7 +577,7 @@ multidict_get(
              *_default = Py_None,
              *ret;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *getone_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:getone",
@@ -833,7 +833,7 @@ multidict_tp_init(MultiDictObject *self, PyObject *args, PyObject *kwds)
 static inline PyObject *
 multidict_add(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -846,7 +846,7 @@ multidict_add(
     PyObject *key = NULL,
              *val = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *kwlist[] = {"key", "value", NULL};
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO:add",
                                      kwlist, &key, &val))
@@ -913,7 +913,7 @@ multidict_clear(MultiDictObject *self)
 static inline PyObject *
 multidict_setdefault(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -926,7 +926,7 @@ multidict_setdefault(
     PyObject *key      = NULL,
              *_default = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *setdefault_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:setdefault",
@@ -967,7 +967,7 @@ skip_optional_pos:
 static inline PyObject *
 multidict_popone(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -981,7 +981,7 @@ multidict_popone(
              *_default = NULL,
              *ret_val  = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *popone_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:popone",
@@ -1046,7 +1046,7 @@ skip_optional_pos:
 static inline PyObject *
 multidict_pop(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -1060,7 +1060,7 @@ multidict_pop(
              *_default = NULL,
              *ret_val  = NULL;
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *pop_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:popone",
@@ -1113,7 +1113,7 @@ skip_optional_pos:
 static inline PyObject *
 multidict_popall(
     MultiDictObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -1128,7 +1128,7 @@ multidict_popall(
              *ret_val  = NULL;
 
 
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     static char *popall_keywords[] = {"key", "default", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O:popall",
@@ -1270,7 +1270,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "getall",
         (PyCFunction)multidict_getall,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1281,7 +1281,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "getone",
         (PyCFunction)multidict_getone,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1292,7 +1292,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "get",
         (PyCFunction)multidict_get,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1321,7 +1321,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "add",
         (PyCFunction)multidict_add,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1350,7 +1350,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "setdefault",
         (PyCFunction)multidict_setdefault,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1361,7 +1361,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "popone",
         (PyCFunction)multidict_popone,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1372,7 +1372,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "pop",
         (PyCFunction)multidict_pop,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1383,7 +1383,7 @@ static PyMethodDef multidict_methods[] = {
     {
         "popall",
         (PyCFunction)multidict_popall,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1559,7 +1559,7 @@ multidict_proxy_tp_init(MultiDictProxyObject *self, PyObject *args,
 static inline PyObject *
 multidict_proxy_getall(
     MultiDictProxyObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -1572,7 +1572,7 @@ multidict_proxy_getall(
     return multidict_getall(
         self->md,
         args,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         kwds
 #else
         nargs,
@@ -1584,7 +1584,7 @@ multidict_proxy_getall(
 static inline PyObject *
 multidict_proxy_getone(
     MultiDictProxyObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -1596,7 +1596,7 @@ multidict_proxy_getone(
 {
     return multidict_getone(
         self->md, args,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         kwds
 #else
         nargs, kwnames
@@ -1607,7 +1607,7 @@ multidict_proxy_getone(
 static inline PyObject *
 multidict_proxy_get(
     MultiDictProxyObject *self,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
     PyObject *args,
     PyObject *kwds
 #else
@@ -1620,7 +1620,7 @@ multidict_proxy_get(
     return multidict_get(
         self->md,
         args,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         kwds
 #else
         nargs,
@@ -1734,7 +1734,7 @@ static PyMethodDef multidict_proxy_methods[] = {
     {
         "getall",
         (PyCFunction)multidict_proxy_getall,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1745,7 +1745,7 @@ static PyMethodDef multidict_proxy_methods[] = {
     {
         "getone",
         (PyCFunction)multidict_proxy_getone,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL
@@ -1756,7 +1756,7 @@ static PyMethodDef multidict_proxy_methods[] = {
     {
         "get",
         (PyCFunction)multidict_proxy_get,
-#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 13
+#if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 12
         METH_VARARGS
 #else
         METH_FASTCALL

--- a/tests/test_multidict.py
+++ b/tests/test_multidict.py
@@ -209,6 +209,10 @@ class BaseMultiDictTest:
 
         assert d.getone("key2", "default") == "default"
 
+    def test_call_with_kwargs(self, cls: Type[MultiDict[str]]) -> None:
+        d = cls([("present", "value")])
+        assert d.getall(default="missing", key="notfound") == "missing"
+
     def test__iter__(
         self,
         cls: Union[

